Automattic/jetpack
# Social: Add share status info

> When the user publishes a post, it will take a few seconds for the share status to become available. 
> 
> Till it becomes available, show the loader: 
> 
> From the time the post gets published, till a maximum period of 1 minute, check every 5 seconds if the post meta becomes available to display the share status. 
> 

## Proposed changes:

* Add a REST endpoint to get the post share status
* Create social data store logic to handle fetching of post share status
* Add the share status info to the Jetpack/Social sidebar

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable the share status feature - `define( 'JETPACK_SOCIAL_HAS_SHARE_STATUS', true );`
* Create a new post and publish it
* Confirm that you see the share status info in the sidebar in post publish panel
* Confirm that it waits for no more than a minute till the status is retrieved

When sharing is successful
<img width="279" alt="image" src="https://github.com/user-attachments/assets/cf9ee21e-7d74-494b-ad63-3efb4d0f63d5">

When there is a failure
<img width="275" alt="image" src="https://github.com/user-attachments/assets/204e5a88-2f3c-41a7-9385-a4541d53a024">

When fetching the share status
<img width="305" alt="image" src="https://github.com/user-attachments/assets/da7196a9-99a6-4062-9db1-e72f7371bb74">

When all the connections are disabled and the post is not shared to any connection
<img width="291" alt="image" src="https://github.com/user-attachments/assets/edf5d067-fc1d-4097-844c-86a068e9db0c">
